lib/ansible/modules/cloud/amazon/route53.py

```diff
@@ -475,9 +475,7 @@ def invoke_with_throttling_retries(function_ref, *argv, **kwargs):
 def decode_name(name):
     # Due to a bug in either AWS or Boto, "special" characters are returned as octals, preventing round
     # tripping of things like * and @.
-    decoded_name = name.replace(r'\052', '*')
-    decoded_name = decoded_name.replace(r'\100', '@')
-    return decoded_name
+    return name.encode().decode('unicode_escape')
```

Comment:
This seems to work in both Py 2 and Py 3 for the examples I tested, but I have no idea whether it works in general (need to do some research).

Is there a reason you need this? For which strings did you have problems?

Reply:
The original implementation only fixes two cases ("*" and "@"), this new one is generic.
I had problems, for example, when I used "#" as prefix to disable some records (it's suppressed from route53, and from Bind, while still a valid record)
Then, instead of adding it, I've generalized the function.
The encode with "unicode_scape" replaces escaped Unicode literals, so it works for all returned values from route53, which are Unicode encoded.

Reply:
I'm mainly wondering whether there is a case where this breaks something :) But if `name` is only using ASCII chars, this should work fine.

I've tested the module with the DNS entries at work, and it doesn't seem to break there. So I'm happy ;)

Could you add a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to)?

Reply:
Changelog fragment added